autoPyTorch/api/base_task.py

```diff
@@ -143,40 +744,100 @@ def predict(
+        except sklearn.exceptions.NotFittedError:
+            # When training a cross validation model, self.cv_models_
+            # will contain the Voting classifier/regressor product of cv
+            # self.models_ in the case of cv, contains unfitted models
+            # Raising above exception is a mechanism to detect which
+            # attribute contains the relevant models for prediction
+            try:
+                check_is_fitted(list(self.cv_models_.values())[0])
+                models = self.cv_models_
+            except sklearn.exceptions.NotFittedError:
+                raise ValueError('Found no fitted models!')
+
+        all_predictions = joblib.Parallel(n_jobs=n_jobs)(
+            joblib.delayed(_pipeline_predict)(
+                models[identifier], X_test, batch_size, self._logger, self.task_type
+            )
+            for identifier in self.ensemble_.get_selected_model_identifiers()
+        )
+
+        if len(all_predictions) == 0:
+            raise ValueError('Something went wrong generating the predictions. '
+                             'The ensemble should consist of the following '
+                             'models: %s, the following models were loaded: '
+                             '%s' % (str(list(self.ensemble_.indices_)),
+                                     str(list(self.models_))))
+
+        predictions = self.ensemble_.predict(all_predictions)
+
+        if self.task_type not in REGRESSION_TASKS:
+            # Make sure average prediction probabilities
+            # are within a valid range
+            # Individual models are checked in _model_predict
+            predictions = np.clip(predictions, 0.0, 1.0)
+
+        self._clean_logger()
+
+        return predictions
 
     def score(
-        self,
-        X_test: np.ndarray,
-        y_test: np.ndarray,
-        sample_weights: Optional[np.ndarray] = None,
-    ) -> float:
+            self,
+            y_pred: np.ndarray,
+            y_test: Union[np.ndarray, pd.DataFrame]
+    ) -> Dict[str, float]:
         """Calculate the score on the test set.
         Calculate the evaluation measure on the test set.
         Args:
-        X_test: (np.ndarray)
-            The test examples of the dataset.
+        y_pred: (np.ndarray)
+            The test predictions
         y_test: (np.ndarray)
             The test ground truth labels.
-        sample_weights: (np.ndarray|None)
-            The weights for each sample.
         Returns:
-            Value of the evaluation metric calculated on the test set.
+            Dict[str, float]: Value of the evaluation metric calculated on the test set.
         """
-        return self.pipeline.score(X_test, y_test, sample_weights)
+        if isinstance(y_test, pd.Series):
+            y_test = y_test.to_numpy(dtype=np.float)
+
+        if self._metric is None:
+            raise ValueError("AutoPytorch failed to infer a metric from the dataset "
```

Comment:
More precise "No metric found. Either fit/search has not been called yet or AutoPyTorch failed to infer a metric from the dataset"